lib/taskana-core/src/main/java/pro/taskana/TaskanaRole.java

```diff
@@ -16,17 +22,17 @@
     }
 
     public static TaskanaRole fromPropertyName(String name) {
-        if (USER.propertyName.equalsIgnoreCase(name)) {
-            return TaskanaRole.USER;
-        } else if (BUSINESS_ADMIN.propertyName.equalsIgnoreCase(name)) {
-            return TaskanaRole.BUSINESS_ADMIN;
-        } else if (ADMIN.propertyName.equalsIgnoreCase(name)) {
-            return TaskanaRole.ADMIN;
-        } else if (MONITOR.propertyName.equalsIgnoreCase(name)) {
-            return TaskanaRole.MONITOR;
-        } else {
-            return null;
-        }
+        return Arrays.stream(TaskanaRole.values())
+            .filter(x -> x.propertyName.equalsIgnoreCase(name))
+            .findFirst()
+            .orElseThrow(() -> new SystemException(
+                "Internal System error when processing role property " + name));
```

Comment:
Why are you throwing an exception here? This is a new behaviour.

Reply:
It was just moved from initTaskanaRoles to prevent returning null. since this is the only method calling that, it does not change anything. 
suggestions? alternatively i could return an optional

Reply:
I saw what you described later during the review. The solution is fine here. Personally I would have thrown an InvalidArgumentException.